From 65404e2a7f2fec0fffe8128bf2f178174957c75e Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Wed, 18 Feb 2026 15:03:15 +0100
Subject: [PATCH] libinput: fix missing native language dependency

Added NLS dependency and fixing compilation error:

libglib-2.0.so.0: undefined reference to `libiconv'
libglib-2.0.so.0: undefined reference to `libiconv_open'
libglib-2.0.so.0: undefined reference to `libintl_bind_textdomain_codeset'
libglib-2.0.so.0: undefined reference to `libiconv_close'
libglib-2.0.so.0: undefined reference to `libintl_dcgettext'
libglib-2.0.so.0: undefined reference to `libintl_textdomain'
libglib-2.0.so.0: undefined reference to `libintl_bindtextdomain'
libglib-2.0.so.0: undefined reference to `libintl_gettext'
libglib-2.0.so.0: undefined reference to `libintl_dgettext'
libglib-2.0.so.0: undefined reference to `libintl_dngettext'
---
 libs/libinput/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libs/libinput/Makefile b/libs/libinput/Makefile
index 12825467c..8880b6596 100644
--- a/libs/libinput/Makefile
+++ b/libs/libinput/Makefile
@@ -18,6 +18,7 @@ PKG_LICENSE_FILES:=COPYING
 PKG_CPE_ID:=cpe:/a:freedesktop:libinput
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
 include $(INCLUDE_DIR)/meson.mk
 
 define Package/libinput
@@ -25,7 +26,7 @@ define Package/libinput
   CATEGORY:=Libraries
   TITLE:=a library to handle input devices
   URL:=https://freedesktop.org/wiki/Software/libinput/
-  DEPENDS:=+libevdev +mtdev +libudev +libwacom
+  DEPENDS:=+libevdev +mtdev +libudev +libwacom $(ICONV_DEPENDS) $(INTL_DEPENDS)
 endef
 
 define Package/libinput/description
-- 
2.53.0

